lib/react_client/react_interop.dart

```diff
@@ -423,8 +425,8 @@ class ReactDartInteropStatics2 implements ReactDartInteropStatics {
 /// See [ReactDartInteropStatics], [createReactDartComponentClass].
 class ComponentStatics<T extends Component> {
   final ComponentFactory<T> componentFactory;
-
-  ComponentStatics(this.componentFactory);
+  final Component2 componentInstance;
```

Comment:
Also... if this is necessary - shouldn't it be 

```suggestion
  final T componentInstance;
```

Reply:
Because both `Component` and `Component2` rely on `ComponentStatics`, this would work but ends up requiring explicitly typing instances of `ComponentStatics` for the sake of differentiating the usages between `Component` and `Component2`. @greglittlefield-wf suggested just implementing `ComponentStatics2` because then this works easily and removes the need for `ComponentStatics` to have `componentInstance`.